staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope.go

```diff
@@ -156,9 +162,10 @@ func (t *envelopeTransformer) getTransformer(encKey []byte) value.Transformer {
 }
 
 // generateKey generates a random key using system randomness.
-func generateKey(length int) ([]byte, error) {
-	key := make([]byte, length)
-	_, err := rand.Read(key)
+func generateKey(length int) (key []byte, err error) {
+	defer value.RecordDataKeyGeneration(time.Now(), err)
```

Comment:
I think this should be
`defer func(start time.Time) { value.RecordDataKeyGeneration(start, err) } (time.Now())`
So that `err` gets evaluated after return

https://play.golang.org/p/9djh_YLNCvO

Reply:
Done